File: backend/FinancialControl.Api/Models/Goal.cs
namespace FinancialControl.Api.Models;

public class Goal
{
    public Guid Id { get; set; }
    public Guid UserId { get; set; }
    public string Name { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public decimal TargetAmount { get; set; }
    public decimal CurrentAmount { get; set; }
    public DateTime TargetDate { get; set; }
    public GoalStatus Status { get; set; }
    public GoalPriority Priority { get; set; }
    public string? ImageUrl { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }

    // Navigation properties
    public User User { get; set; } = null!;
    public ICollection<GoalContribution> Contributions { get; set; } = new List<GoalContribution>();
}

public class GoalContribution
{
    public Guid Id { get; set; }
    public Guid GoalId { get; set; }
    public decimal Amount { get; set; }
    public string? Note { get; set; }
    public DateTime ContributedAt { get; set; }

    // Navigation properties
    public Goal Goal { get; set; } = null!;
}

public enum GoalStatus
{
    Active,
    Completed,
    Cancelled,
    Paused
}

public enum GoalPriority
{
    Low,
    Medium,
    High,
    Critical
}

File: frontend/src/types/goal.ts
export interface Goal {
  id: string
  userId: string
  name: string
  description: string
  targetAmount: number
  currentAmount: number
  targetDate: string
  status: string
  priority: string
  imageUrl?: string
  percentageComplete: number
  remainingAmount: number
  daysRemaining: number
  requiredMonthlyContribution: number
  createdAt: string
  updatedAt: string
  recentContributions: GoalContribution[]
}

export interface CreateGoal {
  name: string
  description: string
  targetAmount: number
  targetDate: string
  priority: string
  imageUrl?: string
}

export interface UpdateGoal {
  name?: string
  description?: string
  targetAmount?: number
  targetDate?: string
  status?: string
  priority?: string
  imageUrl?: string
}

export interface GoalContribution {
  id: string
  goalId: string
  amount: number
  note?: string
  contributedAt: string
}

export interface CreateContribution {
  amount: number
  note?: string
}

export interface GoalSummary {
  totalGoals: number
  activeGoals: number
  completedGoals: number
  totalTargetAmount: number
  totalCurrentAmount: number
  overallProgress: number
  goals: Goal[]
}
